src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/credentials/certificate/AzureCertificateCredentials.java

```diff
@@ -0,0 +1,126 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.azurekeyvaultplugin.AzureCredentialsProvider;
+import org.jvnet.localizer.ResourceBundleHolder;
+
+public class AzureCertificateCredentials extends BaseStandardCredentials implements StandardCertificateCredentials {
+    private static final Logger LOG = Logger.getLogger(AzureCertificateCredentials.class.getName());
+
+    private final Supplier<Secret> keyStoreSource;
+    private final Supplier<Secret> password;
+
+    public AzureCertificateCredentials(
+        CredentialsScope scope,
+        String id,
+        String description,
+        Supplier<Secret> password,
+        Supplier<Secret> keyStoreSource
+    ) {
+        super(scope, id, description);
+
+        Objects.requireNonNull(keyStoreSource);
+        this.password = password;
+        this.keyStoreSource = keyStoreSource;
+    }
+
+    /**
+     * Helper to convert a {@link Secret} password into a {@code char[]}
+     *
+     * @param password the password.
+     * @return a {@code char[]} containing the password or {@code null}
+     */
+    @CheckForNull
+    private static char[] toCharArray(@NonNull Secret password) {
+        String plainText = Util.fixEmpty(password.getPlainText());
+        return plainText == null ? null : plainText.toCharArray();
+    }
+
+    @NonNull
+    @Override
+    public KeyStore getKeyStore()
+    {
+        KeyStore keyStore;
+
+        try {
+            keyStore = KeyStore.getInstance("PKCS12");
+        } catch (java.security.KeyStoreException e) {
+            throw new IllegalStateException("PKCS12 is a keystore type per the JLS spec", e);
+        }
+
+        ByteArrayInputStream keyStoreByteInputStream;
+
+        try{
+            keyStoreByteInputStream = new ByteArrayInputStream(
+                Base64.getDecoder().decode(
+                    Secret.toString(keyStoreSource.get())
+                )
+            );
+        } catch (java.lang.IllegalArgumentException e) {
+            LOG.log(Level.WARNING, "Error decoding Keystore. A base64 encoded certificate is expected. Secret ID:" + getId() + ". " + e.getMessage(), e);
+            throw new IllegalStateException("Cannot decode keystore", e);
+        }
+
+
+        try {
+            keyStore.load(keyStoreByteInputStream, toCharArray(getPassword()));
+        }
+        catch(
+            java.security.cert.CertificateException |
+            java.security.NoSuchAlgorithmException |
+            java.io.IOException e
+        ){
+            LOG.log(Level.WARNING, "Error loading Keystore . Secret ID:" + getId() + ". " + e.getMessage(), e);
+            throw new IllegalStateException("Error loading Keystore.", e);
+        }
+        return keyStore;
+    }
+
+    @NonNull
+    public Secret getKeyStoreSecret() {
+        return keyStoreSource.get();
+    }
+
+    @NonNull
+    @Override
+    public Secret getPassword() {
+        return password.get();
```

Comment:
looks like this could be null if you don't supply a password?

Which is likely a valid case as Azure Key Vault normally strips passwords when you upload certificates and sets them to be an empty string `""`

Reply:
![image](https://github.com/jenkinsci/azure-keyvault-plugin/assets/1141454/2190913d-f256-426f-8886-f5ea480a127c)
adding a certificate credentials to jenkins expects a password value 

but I am not sure if the password can be empty or not ( I did not have that case so far ) 

I haven't implemented the case where a password for the keystore is not needed, I will have a look later

Reply:
![image](https://github.com/jenkinsci/azure-keyvault-plugin/assets/1141454/10563af4-5bf4-4dbf-9311-2f4b1e109c85)
ok, the credentials binding plugin says the password variable is optional 

so I guess yes the password is optional :) will have a look

Reply:
it'll either be optional or an empty string.

Try generate a selfsigned certificate in key vault to check out the workflow.

It looks like it might not be relevant because the credentials-binding must handle the actual writing to disk but this is how the custom step does it:

https://github.com/jenkinsci/azure-keyvault-plugin/blob/master/src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultUtil.java#L70-L100